test/SlowTests/Cluster/ClusterTransactionTests.cs

```diff
@@ -1456,5 +1451,626 @@ select new
+        {
+            public override string IndexName { get; }
+
+            public TestIndex(string name)
+            {
+                IndexName = name;
+                Map = testObjs => from user in testObjs
+                    select new { user.Prop };
+            }
+        }
+
+        private class TestCommand : TransactionOperationsMerger.MergedTransactionCommand
+        {
+            private readonly ManualResetEvent _manualResetEvent;
+            private readonly TimeSpan _timeout;
+
+            public TestCommand(ManualResetEvent manualResetEvent, TimeSpan timeout)
+            {
+                if (Debugger.IsAttached)
+                    timeout *= 10;
+                
+                _manualResetEvent = manualResetEvent;
+                _timeout = timeout;
+            }
+
+            protected override long ExecuteCmd(DocumentsOperationContext context)
+            {
+                _manualResetEvent.WaitOne(_timeout);
+                return 1;
+            }
+
+            public override TransactionOperationsMerger.IReplayableCommandDto<TransactionOperationsMerger.MergedTransactionCommand> ToDto<TTransaction>(
+                TransactionOperationContext<TTransaction> context)
+            {
+                throw new NotImplementedException();
+            }
+        }
+        
+        [RavenTheory(RavenTestCategory.ClusterTransactions)]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task ClusterTransaction_WhenLoadReturnEmptyAndCompareExchangeExit_ShouldStillThrowConcurrency(bool clusterTrxBefore)
+        {
+            const string id = "testObjs/1";
+            using var store = GetDocumentStore();
+
+            if (clusterTrxBefore)
+            {
+                using var session = store.OpenAsyncSession(new SessionOptions { TransactionMode = TransactionMode.ClusterWide });
+                await session.StoreAsync(new TestObj());
+                await session.SaveChangesAsync();
+            }
+
+            using var mre = new ManualResetEvent(false);
+            try
+            {
+                var database = await GetDatabase(store.Database);
+                var longCommandTask = database.TxMerger.Enqueue(new TestCommand(mre, TimeSpan.FromSeconds(15)));
+
+                var halfProcessedTask = Task.Run(async () =>
+                {
+                    using var session = store.OpenAsyncSession(new SessionOptions
+                    {
+                        TransactionMode = TransactionMode.ClusterWide
+                    });
+                    {
+                        await session.StoreAsync(new TestObj{Prop = "new 1"}, id);
+                        await session.SaveChangesAsync();
+                    }
+                });
+                using var session = store.OpenAsyncSession(new SessionOptions { TransactionMode = TransactionMode.ClusterWide });
+
+                await Task.Delay(10);
```

Comment:
you have many `Task.Delay` in those tests, do we really need them all? 
it is okay to have those if you are polling something in a loop, but please go over them and check if they are really needed

Reply:
It is only 10 milliseconds and just to increase the chance for the race condition.
I just wanted it to suspend the task.
What is the issue?

Reply:
Having many `Task.Delay` in tests will make the entire test suit to run longer than necessary and usually for timing it racy thing to do.

This is not the only one that you have.
yes, 10ms is a small value, what race would it expose?